src/Ast/Type/CallableTypeNode.php

```diff
@@ -22,7 +22,7 @@ class CallableTypeNode implements TypeNode
 	/**
 	 * @param CallableTypeParameterNode[] $parameters
 	 */
-	public function __construct(IdentifierTypeNode $identifier, array $parameters, TypeNode $returnType)
+	public function __construct(IdentifierTypeNode|GenericTypeNode $identifier, array $parameters, TypeNode $returnType)
```

Comment:
@ondrejmirtes is this the right OOP design or you want separate `GenericCallableTypeNode`?

`GenericTypeNode` is a litttle problematic then, as `$variances` are stored separately from `$genericTypes` there, shouldn't that be stored in another separate class? /cc @jiripudil